openedx/features/courseware/tests/test_pre_post_assessment_scores_api.py

```diff
@@ -0,0 +1,148 @@
+        url = reverse(
+            'courseware:pre_post_assessments_score',
+            kwargs={
+                'course_id': self.course.id,
+                'chapter_id': chapter_id,
+            }
+        )
+
+        return self.client.get(url)
+
+    def add_pre_post_assessment_grading_policy(self):
+        """Add grading policy to mark assessments pre or post for testing"""
+        grading_policy = {
+            "GRADER": [
+                {
+                    "type": constants.PRE_ASSESSMENT_FORMAT,
+                    "min_count": 1,
+                    "drop_count": 0,
+                    "short_label": constants.PRE_ASSESSMENT_FORMAT,
+                    "weight": 0.0
+                },
+                {
+                    "type": constants.POST_ASSESSMENT_FORMAT,
+                    "min_count": 1,
+                    "drop_count": 0,
+                    "short_label": constants.POST_ASSESSMENT_FORMAT,
+                    "weight": 0.25
+                }
+            ]
+        }
+
+        self.course.grading_policy = grading_policy
+        self.update_course(self.course, self.user.id)
+        self.course = self.store.get_course(self.course.id)
+
+
+class PrePostAssessmentScoreForUnEnrolledUserTest(PrePostAssessmentMixin, APITestCase):
+    """Tests for unenrolled and unauthorized users"""
+
+    def setUp(self):
+        super(PrePostAssessmentScoreForUnEnrolledUserTest, self).setUp()
+        self.create_course_and_user()
+        self.client.force_authenticate(self.user)
+
+    def test_un_enrolled_user_request(self):
+        """Test unenrolled user request"""
+        response = self.get_pre_post_assessment_api_response(self.create_chapter().location.block_id)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_unauthorized_request(self):
+        """Test unauthorized request"""
+        self.client.force_authenticate(user=None)
+        response = self.get_pre_post_assessment_api_response(self.create_chapter().location.block_id)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class PrePostAssessmentScoreForEnrolledUserTest(PrePostAssessmentMixin, APITestCase):
+    """Tests for enrolled users"""
+
+    def setUp(self):
+        super(PrePostAssessmentScoreForEnrolledUserTest, self).setUp()
+        self.create_course_and_user()
+        self.client.force_authenticate(self.user)
+        CourseEnrollmentFactory.create(user=self.user, course_id=self.course.id)
+
+    def test_invalid_chapter_id(self):
+        """Test api for invalid chapter id"""
+        response = self.get_pre_post_assessment_api_response('invalid_chapter_id')
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_no_pre_post_assessment_in_chapter(self):
+        """Test if module/chapter has no pre/post assessment"""
+        chapter = self.create_chapter()
+        ItemFactory.create(
```

Comment:
Do you think, there is a significant ROI if we tackle this creation in `create_assessment_subsection` as well?